common/djangoapps/student/models.py

```diff
@@ -1293,6 +1293,32 @@ class Meta:  # pylint: disable=missing-docstring
     def __unicode__(self):
         return "[CourseEnrollmentAllowed] %s: %s (%s)" % (self.email, self.course_id, self.created)
 
+    @classmethod
+    def may_enroll_in(cls, course_id):
+        """
+        Return QuerySet of students who are allowed to enroll in a course.
+
+        Note that result *includes* students who have already enrolled
+        in the course.
+
+        `course_id` identifies the course for which to compute the QuerySet.
+        """
+        return CourseEnrollmentAllowed.objects.filter(course_id=course_id)
+
+    @classmethod
+    def num_may_enroll_in(cls, course_id):
```

Comment:
@sarina I agree. Would renaming it to `num_may_enroll_but_unenrolled` make it clearer? I only use this method once throughout the rest of the code, but I still feel like it makes sense to encapsulate the logic for computing this number in the model class. What do you think?